op-program/client/driver/driver.go

```diff
@@ -43,9 +43,8 @@ func NewDriver(logger log.Logger, cfg *rollup.Config, depSet derive.DependencySe
 	pipelineDeriver := derive.NewPipelineDeriver(context.Background(), pipeline)
 	pipelineDeriver.AttachEmitter(d)
 
-	ec := engine.NewEngineController(l2Source, logger, metrics.NoopMetrics, cfg, &sync.Config{SyncMode: sync.CLSync}, d)
-	engineDeriv := engine.NewEngDeriver(logger, context.Background(), cfg, metrics.NoopMetrics, ec)
-	engineDeriv.AttachEmitter(d)
+	ec := engine.NewEngineControllerWithEventFields(context.Background(), l2Source, logger, metrics.NoopMetrics, cfg, &sync.Config{SyncMode: sync.CLSync}, d)
```

Comment:
`NewEngineControllerWithEventFields` receives emitter arg `d` and initializes `EngineController` so we do not have to call `AttachEmitter` again.